backend/integrations/gitlab/gitlab.go

```diff
@@ -0,0 +1,365 @@
+	return doGitlabRequest[T]("GET", accessToken, url, "")
+}
+
+func doGitlabRequest[T any](method string, accessToken string, url string, body string) (T, error) {
+	var unmarshalled T
+	client := &http.Client{}
+
+	// code to tell whether we are using absoluteUrl or relative url
+	var finalUrl = fmt.Sprintf("%s%s", GitlabApiBaseUrl, url)
+	parsedUrl, err := nUrl.Parse(url)
+
+	if err != nil {
+		return unmarshalled, errors.Wrap(err, "error parsing url") // this really should not happen
+	}
+
+	if parsedUrl.IsAbs() {
+		finalUrl = url
+	}
+
+	req, err := http.NewRequest(method, finalUrl, strings.NewReader(body))
+	if err != nil {
+		return unmarshalled, errors.Wrap(err, "error creating api request to Gitlab")
+	}
+
+	if accessToken != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	}
+	if method != "GET" {
+		req.Header.Set("Content-Type", "application/json")
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return unmarshalled, errors.Wrap(err, "error getting response from gitlab endpoint")
+	}
+
+	b, err := io.ReadAll(res.Body)
+	if res.StatusCode != 200 && res.StatusCode != 201 {
+		return unmarshalled, errors.New("Gitlab API responded with error; status_code=" + res.Status + "; body=" + string(b))
+	}
+
+	if err != nil {
+		return unmarshalled, errors.Wrap(err, "error reading response body from gitlab endpoint")
+	}
+
+	err = json.Unmarshal(b, &unmarshalled)
+	if err != nil {
+		return unmarshalled, errors.Wrap(err, "error unmarshaling gitlab api response"+string(b))
+	}
+
+	return unmarshalled, nil
+}
+
+type GitlabProjectResponse struct {
+	ID                int       `json:"id"`
+	Description       *string   `json:"description"`
+	Name              string    `json:"name"`
+	NameWithNamespace string    `json:"name_with_namespace"`
+	Path              string    `json:"path"`
+	PathWithNamespace string    `json:"path_with_namespace"`
+	CreatedAt         time.Time `json:"created_at"`
+	DefaultBranch     string    `json:"default_branch"`
+	TagList           []string  `json:"tag_list"`
+	Topics            []string  `json:"topics"`
+	SSHURLToRepo      string    `json:"ssh_url_to_repo"`
+	HTTPURLToRepo     string    `json:"http_url_to_repo"`
+	WebURL            string    `json:"web_url"`
+	ReadmeURL         string    `json:"readme_url"`
+	ForksCount        int       `json:"forks_count"`
+	AvatarURL         *string   `json:"avatar_url"`
+	StarCount         int       `json:"star_count"`
+	LastActivityAt    time.Time `json:"last_activity_at"`
+	Namespace         Namespace `json:"namespace"`
+}
+
+type Namespace struct {
+	ID        int     `json:"id"`
+	Name      string  `json:"name"`
+	Path      string  `json:"path"`
+	Kind      string  `json:"kind"`
+	FullPath  string  `json:"full_path"`
+	ParentID  *int    `json:"parent_id"`
+	AvatarURL *string `json:"avatar_url"`
+	WebURL    string  `json:"web_url"`
+}
+
+func getQueryParams() string {
```

Comment:
lastActivity is an attempt to limit the size of the projects returned. 
This is my way of saying only recently active projects

Reply:
So just to make sure I'm understanding, the list of projects we show will be GitLab projects with an activity in the last 2 years?

It looks like we just get all the repos for GitHub but good with this approach

Reply:
That is right. We get all projects with an activity in the last 2 years.